### PR TITLE
Improve error reporting

### PR DESCRIPTION
**Pull request by psiyengar**
Set appropriate Netconf error information when an error is encountered while processing Netconf RPC requests.

### CODE REVIEW

File: netconf.c [hunk 1]
```diff
@@ -1169,32 +1374,37 @@ static bool
 handle_unlock (struct netconf_session *session, xmlNode * rpc)
 {
     xmlNode *action = xmlFirstElementChild (rpc);
-    xmlNode *error_info, *node;
-    char *msg = NULL;
-    char session_id_str[32];
+    xmlNode *node;
+    nc_error_parms error_parms = NC_ERROR_PARMS_INIT;
+    GString *error_msg = g_string_new (NULL);
```
Comment (tony-vanderpeet) [inline netconf.c]: Does this structure need to be freed at some point? Not all return paths do this I think.

File: netconf.c [hunk 2]
```diff
@@ -1213,42 +1423,53 @@ handle_kill_session (struct netconf_session *session, xmlNode * rpc)
     xmlNode *action = xmlFirstElementChild (rpc);
     xmlNode *node;
     uint32_t kill_session_id = 0;
-    char *msg = NULL;
     struct netconf_session *kill_session = NULL;
     xmlChar* content = NULL;
+    nc_error_parms error_parms = NC_ERROR_PARMS_INIT;
```
Comment (tony-vanderpeet) [inline netconf.c]: I think you might need to be careful about cleanup for all these initialised error structures? Have you run apteryx-netconf tests on a PC with valgrind running?

**Review by tony-vanderpeet — COMMENTED** [review]
I think you have a problem with memory leaks here...

**Comment by psiyengar** [comment]
Thanks, Tony. I forgot to run valgrind. Will try to fix the memory leaks.
The tests on the PC all passed.

**Review by tony-vanderpeet — APPROVED** [review]

**Comment by psiyengar** [comment]
Abandoning PR due to merge issues with master. Will resubmit. Thanks.

**Pull request closed without merging**